Define `__STDC_WANT_IEC_60559_TYPES_EXT__` before including any other header
that might happen to include `<float.h>`.

diff -urN hdf5-1.14.6/src/H5private.h.orig hdf5-1.14.6/src/H5private.h
--- hdf5-1.14.6/src/H5private.h.orig	2025-03-06 15:03:01.838887900 +0100
+++ hdf5-1.14.6/src/H5private.h	2025-03-06 15:28:52.044667000 +0100
@@ -22,6 +22,8 @@
 
 #define __STDC_WANT_LIB_EXT2__ 1  // define for TR 24731-2:2010 extensions (e.g., vasprintf)
 #define _GNU_SOURCE
+/* Define __STDC_WANT_IEC_60559_TYPES_EXT__ for _FloatN support, if available */
+#define __STDC_WANT_IEC_60559_TYPES_EXT__
 
 #include "H5public.h" /* Include Public Definitions    */
 
@@ -37,8 +39,6 @@
 #include <string.h>
 #include <time.h>
 
-/* Define __STDC_WANT_IEC_60559_TYPES_EXT__ for _FloatN support, if available */
-#define __STDC_WANT_IEC_60559_TYPES_EXT__
 #include <float.h>
 #include <math.h>
 
